File: app/lib/api/entity/UserEntity.dart
import 'package:app/api/entity/CityEntity.dart';
import 'package:app/api/entity/enums/UserRole.dart';
import 'package:json_annotation/json_annotation.dart';

part 'UserEntity.g.dart';

@JsonSerializable()
class UserEntity {
  int id;
  String uid;
  String name;
  String surname;
  String phone;
  String? photo;
  String? telegram;
  String? whatsapp;
  UserRole role;
  CityEntity city;
  double wallet;
  bool subscription;

  UserEntity({this.telegram, this.whatsapp, required this.id, required this.uid, required this.name, required this.surname, required this.phone, required this.city, this.photo, required this.role, required this.wallet, required this.subscription});

  factory UserEntity.fromJson(Map<String, dynamic> json) => _$UserEntityFromJson(json);
  Map<String, dynamic> toJson() => _$UserEntityToJson(this);
}

File: server/GNext/src/main/java/com/thedeveloper/gnext/entity/UserEntity.java
package com.thedeveloper.gnext.entity;

import com.fasterxml.jackson.annotation.JsonIgnore;
import com.fasterxml.jackson.annotation.JsonView;
import com.thedeveloper.gnext.enums.UserRole;
import com.thedeveloper.gnext.utils.Views;

import jakarta.persistence.*;
import lombok.Data;
import lombok.ToString;

import java.util.Date;

@Entity
@Table(name = "users")
@Data
@ToString
public class UserEntity {
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Id
    private Long id;
    @Enumerated(EnumType.STRING)
    UserRole role;
    @ManyToOne()
    @JoinColumn(name = "city", nullable = true,referencedColumnName = "name")
    CityEntity city;
    String uid;
    @JsonIgnore
    String notifyToken;
    @Column(unique = true)
    String phone;
    @JsonIgnore
    String password;
    String name;
    String number;
    String surname;
    String photo;
    boolean subscription = false;
    double wallet = 0;
    Date createDate;
    Date subStart;
    String telegram;
    String whatsapp;
}
